fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/IndexKeyValueToPartialRecord.java

```diff
@@ -185,27 +183,27 @@ public boolean copy(@Nonnull Descriptors.Descriptor recordDescriptor, @Nonnull M
             if (value == null) {
                 return true;
             }
-            if (!containingType.equals(recordDescriptor)) {
-                containingType = recordDescriptor;
-                fieldDescriptor = recordDescriptor.findFieldByName(field);
+            Descriptors.FieldDescriptor mutableFieldDescriptor = this.fieldDescriptor;
```

Comment:
Is it possible that this is causes too many small object allocations stressing GC if `this.fieldDescriptor` is _always_ different from the `recordDescriptor` for a given execution?